src/html/mod.rs

```diff
@@ -92,6 +92,19 @@ impl Html {
         }
     }
 
+    /// Fast select open edges that match selector into vector of elements
+    pub fn matches<'a>(&'a self, selector: &'a Selector) -> impl Iterator<Item = ElementRef> + 'a {
+        self.tree.root().traverse().filter_map(|node| match node {
```

Comment:
impl Stream - html.tree instead of traverse here.

Reply:
> @adamreichold @cfvescovo Ok, sat down and added real benchmarks to the fray.
> 
> -- 3 methods to select tested [50-100 samples]
> 
> 1. the html [traverse](https://github.com/spider-rs/spider/blob/ed7ad9ad75860cdcc43eacb66742fbf9a601eb28/spider/src/page.rs#L252)
> 2. using `html.tree` via stream to [walk](https://github.com/spider-rs/spider/blob/ed7ad9ad75860cdcc43eacb66742fbf9a601eb28/spider/src/page.rs#L309).
> 3. using the select and the iterating through the results like a normal
> 
> ![streaming the select results](https://user-images.githubusercontent.com/8095978/222296466-9a5ac800-0221-4d01-a9e8-d6417d8adea7.png).
> 
> Using a stream on html.tree wins the race by 30% when testing `figma.com` which has a small set of nodes.
> 
> If you wrap the output of select in a stream the results do not change.
> 
> The time difference scales, the larger the website. I think it may be worth looking into having a feature that allows it to run via streams.
> 
> Here is the full bench https://github.com/spider-rs/spider/blob/main/benches/parse.rs

forgot `to_async` for the streaming bench. nvm output is about the same on one thread. When I get more time ill look into adding concurrency.